submitit/test_helpers.py

```diff
@@ -98,3 +98,41 @@ def submitit_file() -> Path:
     assert submitit_file() == Path(__file__).parent / "__init__.py"
     # In the job we should import submitit from the snapshot dir
     assert job.result() == tmp_path / "submitit/__init__.py"
+
+
+class FakeInfoWatcherWithTimer(core.InfoWatcher):
+    # pylint: disable=abstract-method
+    def __init__(self, delay_s: int = 60, time_change: int = 5, end_result: str = "failed"):
+        super().__init__(delay_s)
+        self.start_timer = time.time()
+        self.time_change = time_change
+        self.end_result = end_result
+
+    def get_state(self, job_id: str, mode: str = "standard") -> str:
+        now = time.time()
+        if now - self.start_timer < self.time_change:
+            return "pending"
+        elif now - self.start_timer > self.time_change and now - self.start_timer < 2 * self.time_change:
+            return "running"
+        return self.end_result
+
+
+class FakeJobWithTimer(core.Job[core.R]):
+    def __init__(self, job_id, folder, time_change: int = 5, end_result: str = "failed"):
+        super().__init__(job_id=job_id, folder=folder)
+        self.time_change = time_change
+        self.end_result = end_result
+        self.watcher = FakeInfoWatcherWithTimer(time_change=self.time_change, end_result=self.end_result)
+
+
+def test_monitor_jobs(tmp_path: Path) -> None:
+    job: FakeJobWithTimer[int] = FakeJobWithTimer(job_id="12", folder=tmp_path)
+    job2: FakeJobWithTimer[int] = FakeJobWithTimer(
+        job_id="13", folder=tmp_path, time_change=10, end_result="done"
+    )
+
+    start_time = time.time()
+    done, failed = helpers.monitor_jobs([job, job2], 5, test_mode=True)
```

Comment:
use a much smaller sleep time (0.02?) so as to speed up the test. Sparing 5s is always important so as to make the most of the CI